File: Sources/NSColorWell-Extension.m
#import "NSColorWell-Extension.h"

@implementation NSColorWell (Extension)
- (void)saveColorToMutableDictionary:(NSMutableDictionary<NSString*,NSColor*>*)dictionary
{
    dictionary[self.description] = self.color;
}

- (void)restoreColorFromDictionary:(NSDictionary<NSString*,NSColor*>*)dictionary
{
    if ([dictionary.allKeys containsObject:self.description]) {
        self.color = dictionary[self.description];
    }
}

@end

File: Sources/NSColorWell-Extension.swift
@objc extension NSColorWell {
    func saveColor(toMutableDictionary dictionary: NSMutableDictionary) {
        dictionary[self.description] = self.color
    }

    func restoreColor(fromDictionary dictionary: NSMutableDictionary) {
        if let color = dictionary[self.description] as? NSColor {
            self.color = color
        }
    }
}
